File: spark/maxTemp/maxTemp.py
import sys

from pyspark import SparkContext, SparkConf

if __name__ == "__main__":
    sc = SparkContext("local", "Pyspark max temperature")
    
    # read
    lines = sc.textFile("/input/maxTemp.txt").map(lambda line: line.split(" "))

    # mapper and reducer
    result = lines.map(lambda x: (str(x[0]), int(x[1]))).reduceByKey(lambda x, y: x if x > y else y)

    # output
    result.saveAsTextFile("/output/maxTempOutput")

File: Scala/maxTemp/maxTemp.scala
val textFile = sc.textFile("/input/maxTempInput.txt")
val records = textFile.map(line => line.split(","))

// mapper
val mapper = records.map(record =>(record(0), record(1).toInt))

// reducer
val maxTemp = mapper.reduceByKey{case(a,b) => if(a > b) {a} else {b}}

maxTemp.saveAsTextFile("out")
